mne/time_frequency/csd.py

```diff
@@ -255,3 +220,201 @@ def compute_epochs_csd(epochs, mode='multitaper', fmin=0, fmax=np.inf,
                                              n_fft=n_fft))
         return csds
+
+
+@verbose
+def csd_array(X, sfreq, mode='multitaper', fmin=0, fmax=np.inf,
+              fsum=True, n_fft=None, mt_bandwidth=None,
+              mt_adaptive=False, mt_low_bias=True, verbose=None):
+    """Estimate cross-spectral density from n_trials x n_series x n_times
+       ndarray.
+
+    Note: Results are scaled by sampling frequency for compatibility with
+          Matlab.
+
+    Parameters
+    ----------
+    X : numpy.ndarray or list of numpy.ndarray
+        The time series data.
+    sfreq : float
+        Sampling frequency of observations.
+    mode : str
+        Spectrum estimation mode can be either: 'multitaper' or 'fourier'.
+    fmin : float
+        Minimum frequency of interest.
+    fmax : float | np.inf
+        Maximum frequency of interest.
+    fsum : bool
+        Sum CSD values for the frequencies of interest. Summing is performed
+        instead of averaging so that accumulated power is comparable to power
+        in the time domain. If True, a single CSD matrix will be returned. If
+        False, the output will be a numpy.ndarray of CSD matrices.
+    n_fft : int | None
+        Length of the FFT. If None the exact number of samples between tmin and
+        tmax will be used.
+    mt_bandwidth : float | None
+        The bandwidth of the multitaper windowing function in Hz.
+        Only used in 'multitaper' mode.
+    mt_adaptive : bool
+        Use adaptive weights to combine the tapered spectra into PSD.
+        Only used in 'multitaper' mode.
+    mt_low_bias : bool
+        Only use tapers with more than 90% spectral concentration within
+        bandwidth. Only used in 'multitaper' mode.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    csd : numpy.ndarray
+        The computed cross spectral-density (either summed or not).
+    freqs : numpy.ndarray
+        Frequencies the cross spectral-density is evaluated at.
+    """
+
+    # Check correctness of input data and parameters
+    if fmax < fmin:
+        raise ValueError('fmax must be larger than fmin')
+
+    if isinstance(X, np.ndarray):
+        if X.ndim == 3:
+            n_trials, n_series, n_times = X.shape
+        elif X.ndim == 2:
+            n_trials = 1
+            n_series, n_times = X.shape
+        else:
+            err_str = "X must be either n_trials x n_seris x n_times "
+            err_str += "or n_series x n_times ndarray"
+            raise ValueError(err_str)
+    elif isinstance(X, list):
+        if any([not isinstance(x, np.ndarray) for x in X]):
+            raise ValueError("All arrays in X must be ndarrays")
+        shapes = set(x.shape for x in X)
+        if len(shapes) != 1:
+            raise ValueError("All arrays in X must be the same shape")
+        n_trials = len(X)
+        n_series, n_times = X[0].shape
```

Comment:
This all looks a bit complicated. How about instead:
```
X = np.asarray(X, dtype=float)  # this will ensure all sizes work
if X.ndim != 3:
    raise ValueError(...)
```
If a user wants to use a single trial they can expand axes with `np.newaxis`.